src/audio/audio_manager.cpp

```diff
@@ -16,17 +21,74 @@ AudioManager::AudioManager() : muted(true) {
         if (!music.openFromFile(filename)) {
             std::cerr << "AudioManager: Cannot open " << filename << std::endl;;
         }
-        music.setLoop(true);
-        if (!muted)
+
+        music.setVolume(d.volume);
+        music.setLoop(d.loop);
+
+        if (!muted) {
             music.play();
+        }
+    });
+
+    events::sound_effects_event.connect([&](const events::AudioData &d) {
+        std::string filename = d.filename;
+
+        sounds[filename]->setVolume(d.volume);
+        sounds[filename]->setLoop(d.loop);
+
+        if (!muted) {
+            sounds[filename]->play();
+        }
+        else if (muted && d.loop) {
+            // Change state to paused so that when audio is un-muted, the looped sound will play
```

Comment:
For one time sounds, this will not be the case. This is for continuous sounds like footsteps. Say we are walking and we mute it and while still walking, unmute it; we will still want to hear the footsteps being played. The only time we don't want to hear it is when we stop walking, which would fire a stop sound effects event.